### PR TITLE
feat(event-bus): log subscription errors

### PR DESCRIPTION
**Pull request by Sikora00**
Log errors produced by an event handler
and by a saga as they crash the app.

Refs #409

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: 409

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by Sikora00** [comment]
It's a revert from the old one https://github.com/nestjs/cqrs/pull/740

**Comment by maurolucc** [comment]
When are you planning to merge this fix? We're eager to enjoy it.

**Review by Tony133 — APPROVED** [review]

File: src/event-bus.ts [hunk 1]
```diff
@@ -27,11 +27,13 @@ export type EventHandlerType<EventBase extends IEvent = IEvent> = Type<
...
 export class EventBus<EventBase extends IEvent = IEvent>
   extends ObservableBus<EventBase>
-  implements IEventBus<EventBase>, OnModuleDestroy {
+  implements IEventBus<EventBase>, OnModuleDestroy
+{
   protected getEventId: (event: EventBase) => string;
   protected readonly subscriptions: Subscription[];
 
   private _publisher: IEventPublisher<EventBase>;
+  private _logger = new Logger('EventBus');
```
Comment (kamilmysliwiec) [inline src/event-bus.ts:36]:
```suggestion
  private readonly _logger = new Logger(EventBus.name);
```

File: src/event-bus.ts [hunk 2]
```diff
@@ -68,7 +70,17 @@ export class EventBus<EventBase extends IEvent = IEvent>
 
   bind(handler: IEventHandler<EventBase>, id: string) {
     const stream$ = id ? this.ofEventId(id) : this.subject$;
-    const subscription = stream$.subscribe((event) => handler.handle(event));
+    const subscription = stream$
+      .pipe(mergeMap((event) => from(handler.handle(event))))
+      .subscribe({
+        error: (error) => {
+          this._logger.error(
+            `${handler.constructor.name} produced an error.`,
```
Comment (kamilmysliwiec) [inline src/event-bus.ts:78]:
```suggestion
            `"${handler.constructor.name}" has thrown an error.`,
```

File: src/event-bus.ts [hunk 3]
```diff
@@ -118,8 +133,19 @@ export class EventBus<EventBase extends IEvent = IEvent>
...
-      .pipe(filter((e) => !!e))
-      .subscribe((command) => this.commandBus.execute(command));
+      .pipe(
+        filter((e) => !!e),
+        mergeMap((command) => from(this.commandBus.execute(command))),
+      )
+      .subscribe({
+        error: (error) => {
+          this._logger.error(
+            `CommandHandler executed by Saga produced an error.`,
```
Comment (kamilmysliwiec) [inline src/event-bus.ts:143]:
```suggestion
            `Command handler which execution was triggered by Saga has thrown an error.`,
```

**Pull request merged**